server/broadcast.go

```diff
@@ -346,12 +356,28 @@ func transcodeSegment(cxn *rtmpConnection, seg *stream.HLSSegment, name string,
 
 	// send segment to the orchestrator
 	glog.V(common.DEBUG).Infof("Submitting segment nonce=%d manifestID=%s seqNo=%d orch=%s", nonce, cxn.mid, seg.SeqNo, sess.OrchestratorInfo.Transcoder)
+	if sess.Sender != nil {
+		if err := sess.Sender.ValidateTicketParams(pmTicketParams(sess.OrchestratorInfo.TicketParams)); err != nil {
+			if err != pm.ErrTicketParamsExpired {
+				cxn.sessManager.removeSession(sess)
+				return nil, err
+			}
 
+			glog.V(common.VERBOSE).Infof("ticket params expired, refreshing for orch=%v", sess.OrchestratorInfo.Transcoder)
+			newSess, err := refreshSession(sess)
+			if err != nil {
+				cxn.sessManager.removeSession(sess)
+				return nil, fmt.Errorf("unable to refresh ticket params for orch=%v err=%v", sess.OrchestratorInfo.Transcoder, err)
+			}
+			sess = newSess
+		}
+	}
 	res, err := SubmitSegment(sess, seg, nonce)
 	if err != nil || res == nil {
+		glog.Errorf("unable to submit segment, removing session for orch=%v err=%v", sess.OrchestratorInfo.Transcoder, err)
```

Comment:
Can I then close the issue I made regarding logging this in processsegment ?

Reply:
Given that the current convention in both `transcodeSegment()` and `SubmitSegment()` is to log error messages in place instead of passing errors to their callers, I think we should stick with that convention for now - I'd be ok with closing the issue for now and if there is a strong desire to change this convention at a later point in time then the case for doing so can be made in a new issue then.